internal/pkg/template/templates/cicd/partials/actions.yml

```diff
@@ -0,0 +1,67 @@
+{{- range $stage := .Stages}}
+{{- range $action := $stage.PreDeployments}}
+Pre{{logicalIDSafe $stage.Name}}DeploymentAction{{logicalIDSafe $action.Name}}BuildProjectRole:
+  Type: AWS::IAM::Role
+  Properties:
+    Path: /
+{{ include "role-config" $ | indent 4}}
+    {{- if $.PermissionsBoundary }}
+    PermissionsBoundary: !Sub 'arn:${AWS::Partition}:iam::${AWS::AccountId}:policy/{{$.PermissionsBoundary}}'
+    {{- end }}
+    Policies:
+      - PolicyName: assume-env-manager
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+          - Effect: Allow
+            Resource: 'arn:aws:iam::{{$stage.AccountID}}:role/{{$.AppName}}-{{$stage.Name}}-EnvManagerRole'
+            Action:
+              - sts:AssumeRole
+      - PolicyName: build-role-policy
+        PolicyDocument:
+{{ include "role-policy-document" $ | indent 10 }}
+
+Pre{{logicalIDSafe $stage.Name}}DeploymentAction{{logicalIDSafe $action.Name}}:
+  Type: AWS::CodeBuild::Project
+  Properties:
+    EncryptionKey: !ImportValue {{$.AppName}}-ArtifactKey
+    ServiceRole: !GetAtt Pre{{logicalIDSafe $stage.Name}}DeploymentAction{{logicalIDSafe $action.Name}}BuildProjectRole.Arn
+    Artifacts:
+      Type: CODEPIPELINE
+{{ include "action-config" $action | indent 4}}
+{{- end}}
+{{- end}}
+
+{{- range $stage := .Stages}}
+{{- range $action := $stage.PostDeployments}}
+Post{{logicalIDSafe $stage.Name}}DeploymentAction{{logicalIDSafe $action.Name}}BuildProjectRole:
```

Comment:
It seems like the pre and post action build project role are pretty much the same. Can we consolidate them?

Reply:
Do you mean have another partial "action-role.yml"? And in here we'd have ⬇️ ? 🤔 
```
Ppre{{logicalIDSafe $stage.Name}}DeploymentAction{{logicalIDSafe $action.Name}}BuildProjectRole:
  {{- include "action-role" $stage | indent 2}}
Post{{logicalIDSafe $stage.Name}}DeploymentAction{{logicalIDSafe $action.Name}}BuildProjectRole:
  {{- include "action-role" $stage | indent 2}} 
```

Reply:
No i meant "do we need pre and post to be separate role?" or can we create one role per action for both pre and post.

Reply:
Ah I see! Umm I feel like different action may require different permissions. For example, a db-migration action might need certain RDS permissions, or to assume a role that has the RDS permissions  (this is not always the EnvManagerRole, as they _could_ have an RDS that's outside of Copilot)...

> can we create one role per action for both pre and post.

Pre/post actions do not come in pair, i.e. I could have one pre-action that does db migration, and one post-action that does integration test - these two actions are independent of each other. 

There are several options here:
1. Share a role among all pre/post actions for now, since all of them require the same permissions - for now. Later, when people can customize the permissions, we split them up. 
2. Do not share the same role.

I ended up choosing the second, because I think even though there isn't a great manifest-based way to customize the role yet, they could easily modify  the role via. YAML patch/CDK override. With the first approach, they'd have create a new role, customize it, and change the role reference from the project.

What do you think!